test/angular/component/component-test-typescript-emitter.spec.js

```diff
@@ -0,0 +1,845 @@
+   expect(this.color.classList.contains(className)).toBeTrue();
+}
+
+public colorShouldNotHaveClass(className): void  {
+   expect(this.color.classList.contains(className)).toBeFalse();
+}
+`);
+    });
+
+    it('should emit for nested element', () => {
+      expect(emitter.emitHasClassAssertion(nestedElement())).to.equal(`public colorShouldHaveClass(rowIndex1, className): void  {
+   expect(this.items[rowIndex1].querySelector('.color').classList.contains(className)).toBeTrue();
+}
+
+public colorShouldNotHaveClass(rowIndex1, className): void  {
+   expect(this.items[rowIndex1].querySelector('.color').classList.contains(className)).toBeFalse();
+}
+`);
+    });
+  });
+
+  describe('selected assertion', () => {
+    it('should emit for simple element', () => {
+      expect(emitter.emitSelectedAssertion(simpleElement())).to.equal(`public colorShouldBeSelected(): void  {
+   expect(this.color.selected).toBeTrue();
+}
+
+public colorShouldNotBeSelected(): void  {
+   expect(this.color.selected).toBeFalse();
+}
+`);
+    });
+
+    it('should emit for nested element', () => {
+      expect(emitter.emitSelectedAssertion(nestedElement())).to.equal(`public colorShouldBeSelected(rowIndex1): void  {
+   expect(this.items[rowIndex1].querySelector('.color').selected).toBeTrue();
+}
+
+public colorShouldNotBeSelected(rowIndex1): void  {
+   expect(this.items[rowIndex1].querySelector('.color').selected).toBeFalse();
+}
+`);
+    });
+  });
+
+  describe('selected by index assertion', () => {
+    it('should emit for simple element', () => {
+      expect(emitter.emitSelectedByIndexAssertion(simpleElement(true))).to.equal(`public colorByIndexShouldBeSelected(index): void  {
+   expect(this.color[index].selected).toBeTrue();
+}
+
+public colorByIndexShouldNotBeSelected(index): void  {
+   expect(this.color[index].selected).toBeFalse();
+}
+`);
+    });
+
+    it('should emit for nested element', () => {
+      expect(emitter.emitSelectedByIndexAssertion(nestedElement(true))).to.equal(`public colorByIndexShouldBeSelected(rowIndex1, index): void  {
+   expect(this.items[rowIndex1].querySelectorAll('.color')[index].selected).toBeTrue();
+}
+
+public colorByIndexShouldNotBeSelected(rowIndex1, index): void  {
+   expect(this.items[rowIndex1].querySelectorAll('.color')[index].selected).toBeFalse();
+}
+`);
+    });
+  });
+
+  describe('selected by value assertion', () => {
+    it('should emit for simple element', () => {
+       expect(emitter.emitSelectedByValueAssertion(simpleElement(true))).to.equal(`public colorByValueShouldBeSelected(value): void  {
+   if (this.color.selected) {
+       expect(this.color.value).toEqual(value);
+   }
+   throw new Error('Element is not selected');
+}
+
+public colorByValueShouldNotBeSelected(value): void  {
+   if (this.color.selected) {
+       expect(this.color.value).not.toEqual(value);
+   }
+   throw new Error('Element is not selected');
```

Comment:
I don't think we need the error here. If we get here, it means the option is not selected, which is what we want to verify